src/pages/workspaces/workspace/RequestAccessModal.js

```diff
@@ -0,0 +1,95 @@
+import _ from 'lodash/fp'
+import { useState } from 'react'
+import { div, h, table, tbody, td, th, thead, tr } from 'react-hyperscript-helpers'
+import { buttonPrimary } from 'src/components/common'
+import { centeredSpinner } from 'src/components/icons'
+import Modal from 'src/components/Modal'
+import { Ajax, useCancellation } from 'src/libs/ajax'
+import { withErrorReporting } from 'src/libs/error'
+import { useOnMount, withBusyState } from 'src/libs/utils'
+
+
+export const RequestAccessModal = ({ onDismiss, workspace }) => {
+  const [groups, setGroups] = useState([])
+  const [accessInstructions, setAccessInstructions] = useState([])
+  const [loadingGroups, setLoadingGroups] = useState(false)
+  const [loadingAccessInstructions, setLoadingAccessInstructions] = useState(false)
+  const signal = useCancellation()
+
+  const { Groups, Workspaces } = Ajax(signal)
+
+  const fetchGroups = _.flow(
+    withBusyState(setLoadingGroups),
+    withErrorReporting('Error loading groups')
+  )(async () => {
+    setGroups(await Groups.list())
+  })
+
+  const fetchAccessInstructions = _.flow(
+    withBusyState(setLoadingAccessInstructions),
+    withErrorReporting('Error loading groups')
+  )(async () => {
+    const i = await Workspaces.workspace(workspace.workspace.namespace, workspace.workspace.name).accessInstructions()
+    console.log(i)
+    setAccessInstructions(i)
+  })
+
+  useOnMount(() => {
+    fetchGroups()
+    fetchAccessInstructions()
+  })
+
+  const loading = loadingGroups || loadingAccessInstructions
+  const groupNames = _.map('groupName', groups)
+  const authDomain = workspace.workspace.authorizationDomain
+  return h(Modal, {
+    title: 'Request Access',
+    width: '40rem',
+    onDismiss
+  }, [
+    div(['You cannot access this Workspace because it contains restricted data. You need permission from the admin(s) of all of the Groups in the Authorization Domain protecting the workspace.']),
```

Comment:
Great suggestions! As I've done before, I just copied this text directly from FireCloud and didn't think about how it could be improved. I suspect the reason it's so similar to the tooltip text is that the tooltip text came from the FC modal text. I'm going to simplify the tooltip text and be more wordy in the modal text.